uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -1112,12 +1112,98 @@ public function testFindOverlappingLeaveRequestsForOneOverlappingLeaveRequest()
     $startDate = '2016-11-01';
     $endDate = '2016-11-03';
 
-    $overlappingRequests = LeaveRequest::findOverlappingLeaveRequests($contactID, $startDate, $endDate);
+    $overlappingRequests = LeaveRequest::findOverlappingLeaveRequests($contactID, $startDate, $endDate, $this->absenceType->id, LeaveRequest::REQUEST_TYPE_LEAVE);
     $this->assertCount(1, $overlappingRequests);
     $this->assertInstanceOf(LeaveRequest::class, $overlappingRequests[0]);
     $this->assertEquals($leaveRequest1->id, $overlappingRequests[0]->id);
   }
 
+  public function testFindOverlappingLeaveRequestsForHoursUnitRequests() {
+    $contactID = 1;
+
+    $testFromDate = '2018-04-13 12:00:00';
+    $testToDate = '2018-04-15 15:00:00';
+
+    $absenceTypes = [
+      'days' => AbsenceTypeFabricator::fabricate([ 'calculation_unit' => 1 ]),
+      'hours' => AbsenceTypeFabricator::fabricate([ 'calculation_unit' => 2 ])
+    ];
+
+    // Test suites: date from, date to, unit, should overlap or not
+    $overlappingRequestsTestSuites = [
```

Comment:
This should be moved to a data provider.

Reply:
OK, we agreed that it should actually stay in the suite - this provides an instant overview of which cases are tested.